File: plugins/plugin-elizacloud/python/elizaos_plugin_elizacloud/models/image.py
from elizaos_plugin_elizacloud.providers import ElizaCloudClient
from elizaos_plugin_elizacloud.types import (
    ElizaCloudConfig,
    ImageDescriptionParams,
    ImageDescriptionResult,
    ImageGenerationParams,
)


async def handle_image_generation(
    config: ElizaCloudConfig,
    params: ImageGenerationParams,
) -> list[dict[str, str]]:
    async with ElizaCloudClient(config) as client:
        return await client.generate_image(params)


async def handle_image_description(
    config: ElizaCloudConfig,
    params: ImageDescriptionParams | str,
) -> ImageDescriptionResult:
    async with ElizaCloudClient(config) as client:
        return await client.describe_image(params)

File: plugins/plugin-elizacloud/rust/src/models/image.rs
#![allow(missing_docs)]

use crate::error::Result;
use crate::providers::client::{ElizaCloudClient, ImageDescriptionInput, ImageResult};
use crate::types::{ElizaCloudConfig, ImageDescriptionResult, ImageGenerationParams};

pub async fn handle_image_generation(
    config: ElizaCloudConfig,
    params: ImageGenerationParams,
) -> Result<Vec<ImageResult>> {
    let client = ElizaCloudClient::new(config)?;
    client.generate_image(params).await
}

pub async fn handle_image_description<T: Into<ImageDescriptionInput>>(
    config: ElizaCloudConfig,
    input: T,
) -> Result<ImageDescriptionResult> {
    let client = ElizaCloudClient::new(config)?;
    client.describe_image(input.into()).await
}
